docs/concepts/pooling.md

```diff
@@ -0,0 +1,111 @@
+
+Users can then create a `Pool`, specifying which drives and which nodes should be part of it.
+
+An Ondat controller will then create a `StorageClass` for this pool. Any application that uses this `StorageClass` will then utilize only the drives in that `Pool`.  
+
+### How to use Storage Pooling
+
+#### Preparing your Drives
+
+Drives will need to be added to the desired Kubernetes nodes and mounted inside `/var/lib/storageos/`. Note: any drives mounted which conform to this pattern `/var/lib/storageos/data/dev[0-9]+` will also be used by volumes that are not a part of a storage pool.
+
+Example:
+If you install an nvme drive and want it to only be used by a storage pool you may mount it to `/var/lib/storageos/nvme1`.
+
+If you install an nvme drive and want it to be used by any volume you may mount it to `/var/lib/storageos/data/dev2`. Drives mounted like this can still be used by storage pools.
+
+#### Creating a Storage Pool
+
+Storage pools are a [Kubernetes custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) and can be applied as such. The custom resource has a few fields that should be supplied.
+
+The storage pool controller will create a `StorageClass` from the storage pool, so the standard storage class fields should be set (for example `allowVolumeExpansion`, `parameters`, `reclaimPolicy` and `volumeBindingMode`). These fields will fallback to their defaults if not set.
+
+The other field that must be set is the `nodeDriveMap`, this defines the drives that are in the storage pool and specifies which nodes they're on and where they're mounted.
+
+Example:
+
+```
+apiVersion: api.storageos.com/v1
+kind: Pool
+metadata:
+  name: "my-fast-storage-pool"
+spec:
+  nodeDriveMap: 
+    worker-1: 
+    - "/var/lib/storageos/nvme1"
+    - "/var/lib/storageos/nvme2"
+    worker-2: 
+    - "/var/lib/storageos/really-fast-nvme"
+    worker-3:
+    - "/var/lib/storageos/data/dev1"
+  volumeBindingMode: Immediate
+  allowVolumeExpansion: true
+  parameters:
+    csi.storage.k8s.io/fstype: ext4
+    storageos.com/replicas: 1
+```
+
+Above defines a storage pool that has 4 drives total, 2 on worker-1 (mounted to `/var/lib/storageos/nvme1` and `/var/lib/storageos/nvme2` respectively), 1 on worker-2 (mounted to `/var/lib/storageos/really-fast-nvme`) and 1 on worker-3 (mounted to `/var/lib/storageos/data/dev1`).
+
+The resulting `StorageClass` will by default use `ext4`, allow volume expansion, immediately bind PVs and have 1 replica. The drive on worker-3 will be used for all volumes that land on that node, as well as any volumes that use the storage pool.
+
+> Note: Volumes still need to be replicated to utilise more than a single node's drives.
+
+#### Using a Storage Pool
+
+When a storage pool is created Ondat's storage pool controller will create a `StorageClass`, named in the form `storageos-<namespace>-<storage pool name>`.
```

Comment:
The kubernetes storageClasses are not namespaced, so I'd like to understand the reasoning behind the `-namespace-` in the storageclass. I'm wondering if that means that I can only create PVCs on the pool if they match Namespace?

My understanding is that Pools should be clusterwide as nodes, storageclasses and PVs are. Otherwise, we would have to create a Pool for the same drives on every namespace that uses that Pool, which the user doesn't necessarily know at the pool provisioning time. 
I might be understanding it wrong, so it would be great to clarify.

Reply:
> The kubernetes storageClasses are not namespaced, so I'd like to understand the reasoning behind the -namespace- in the storageclass. I'm wondering if that means that I can only create PVCs on the pool if they match Namespace?

Generally there's not any amazing reason for them being namespaced, except that it may be useful for customers to have pools of the same name in separate namespaces. It's mostly like this due to it's implementation

>  I'm wondering if that means that I can only create PVCs on the pool if they match Namespace?
Nope, that's not the case. PVCs will still specify the storage class to use which will not be namespaced.